### DISCUSSION TITLE
NatTable UI System Testing

### DISCUSSION THREAD

**Discussion by Greglar4** [eclipse-nattable/nattable#37] [Q&A] [answered]
Hello,

I was wondering if anyone has found a solution for UI system testing of NatTables in a RCP project. We have tried RCPTT and SWTBot and neither can interact with our NatTables in a satisfactory way.

Thanks,

Louis

**Comment by fipro78** [comment, marked as answer]
Hi,

I can only answer from a NatTable project perspective. AFAIK

- SWTBot has a NatTable extension that should work. Not sure how good it is.
- RCPTT started some NatTable support activities, but they did not finalize or finish it. I don't know if this has changed and what the current state is.
- NatTable internally implemented a `SWTUtils` class which is similar to what SWTBot is doing. But at a very basic level, surely not usable in a user project.

What concrete issues do you see in your testing scenario?

Greez,
Dirk

**Reply by Greglar4** [reply]
Ah! We did not realize there was an SWTBot extension specifically for NatTable, it doesn't seem to be available from the eclipse update site we used to get SWTBot. We'll give that a try, thanks!

Louis
